ssz/src/main/java/org/ethereum/beacon/ssz/Hasher.java

```diff
@@ -0,0 +1,77 @@
+package org.ethereum.beacon.ssz;
+
+import net.consensys.cava.bytes.Bytes;
+import org.ethereum.beacon.ssz.ConstructorObjCreator;
+import org.ethereum.beacon.ssz.SSZAnnotationSchemeBuilder;
+import org.ethereum.beacon.ssz.SSZCodecHasher;
+import org.ethereum.beacon.ssz.SSZCodecResolver;
+import org.ethereum.beacon.ssz.SSZHashSerializer;
+import org.ethereum.beacon.ssz.SSZModelCreator;
+import org.ethereum.beacon.ssz.SSZModelFactory;
+import org.ethereum.beacon.ssz.SSZSerializer;
+import org.ethereum.beacon.ssz.SSZSerializerBuilder;
+import org.ethereum.beacon.ssz.SettersObjCreator;
+import tech.pegasys.artemis.util.bytes.BytesValue;
+import java.util.function.Function;
+
+/**
+ * <p>SSZ Hashing helper made according to the following specs:
+ * <a href="https://github.com/ethereum/eth2.0-specs/blob/master/specs/simple-serialize.md#tree-hash">SSZ Tree Hash</a></p>
+ *
+ * <p>It's similar to {@link org.ethereum.beacon.ssz.SSZHasher}, but made to work with BytesValue</p>
+ */
+public class Hasher {
```

Comment:
Maybe stay with `SSZHasher` only then? Don't think we need both.

Reply:
done